src/main/java/org/jitsi/jicofo/jigasi/TranscriberManager.java

```diff
@@ -0,0 +1,292 @@
+    }
+
+    /**
+     * Initialise the manager by starting to listen for
+     * {@link ChatRoomMemberPropertyChangeEvent}s.
+     */
+    public void init()
+    {
+        chatRoom.addMemberPropertyChangeListener(this);
+
+        logger.info("initialised transcriber manager");
+    }
+
+    /**
+     * Stop the manager by stopping to listen for
+     * {@link ChatRoomMemberPropertyChangeEvent}s.
+     */
+    public void dispose()
+    {
+        chatRoom.removeMemberPropertyChangeListener(this);
+
+        logger.info("disposed transcriber manager");
+    }
+
+    /**
+     * Listener method for each {@link ChatRoomMemberPropertyChangeEvent}, which
+     * can possibly indicate the transcriber needs to be invited.
+     *
+     * {@inheritDoc}
+     */
+    @Override
+    public void chatRoomPropertyChanged(ChatRoomMemberPropertyChangeEvent event)
+    {
+        if(event.getNewValue() instanceof Presence)
+        {
+            onNewPresence(event);
+        }
+    }
+
+    /**
+     * Check whether a {@link ChatRoomPropertyChangeEvent} is due to a new
+     * {@link Presence}, and when it is, deal with the information is has
+     * provided.
+     *
+     * @param event the event to check the {@link Presence} off
+     */
+    private void onNewPresence(ChatRoomMemberPropertyChangeEvent event)
+    {
+        Presence presence = getPresenceOrNull(event);
+
+        if(presence == null)
+        {
+            return;
+        }
+
+        if(containsTranscriptionStatus(presence))
+        {
+            maybeStoppedTranscribing(presence);
+        }
+        else if(isRequestingTranscriber(presence) && !active)
+        {
+            startTranscribing();
+        }
+    }
+
+    /**
+     * Check whether the given {@link Presence} contains a
+     * {@link TranscriptionStatusExtension}
+     *
+     * @param p the given {@link Presence} to check
+     * @return true when it contains the extension, false otherwise
+     */
+    private boolean containsTranscriptionStatus(Presence p)
+    {
+        return p.hasExtension(
+            TranscriptionStatusExtension.ELEMENT_NAME,
+            TranscriptionStatusExtension.NAMESPACE
+        );
+    }
+
+    /**
+     * Method which is able to invite the transcriber by dialing Jigasi
+     */
+    private synchronized void startTranscribing()
+    {
+        logger.info("Attempting to invite transcriber");
+
+        Jid jigasiJid = jigasiDetector.selectJigasi();
+
+        RayoIqProvider.DialIq dialIq = new RayoIqProvider.DialIq();
+        dialIq.setDestination("jitsi_meet_transcribe");
+        dialIq.setTo(jigasiJid);
+        dialIq.setType(IQ.Type.set);
+        dialIq.setHeader("JvbRoomName", chatRoom.getName());
+
+        try
+        {
+            IQ response = this.connection.sendPacketAndGetReply(dialIq);
+
+            if (response != null)
+            {
+                if (response.getError() == null)
+                {
+                    active = true;
```

Comment:
We should handle the case where jigasi received the request but failed to join the conference. Perhaps we can leave this for a future improvement